rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -208,7 +211,28 @@ public void receiveHeaders(BtcBlock[] headers) throws IOException, BlockStoreExc
      * Receives only one header of serialized Bitcoin block headers and adds them to the internal BlockChain structure.
      * @param header The bitcoin headers
      */
-    public Integer receiveHeader(BtcBlock header) throws IOException {
+    public Integer receiveHeader(BtcBlock header) throws IOException, BlockStoreException {
+
+        long diffTimeStamp = bridgeConstants.getMinSecondsBetweenCallsToReceiveHeader();
+
+        long currentTimeStamp = rskExecutionBlock.getTimestamp(); //in seconds
+        Optional<Long> optionalLastTimeStamp = provider.getReceiveHeadersLastTimestamp();
+        if (optionalLastTimeStamp.isPresent() && (currentTimeStamp - optionalLastTimeStamp.get().longValue() < diffTimeStamp)) {
+            logger.warn("Receive header last TimeStamp less than {} milliseconds", diffTimeStamp);
+            return RECEIVE_HEADER_CALLED_TOO_SOON;
+        }
+        provider.setReceiveHeadersLastTimestamp(currentTimeStamp);
```

Comment:
ok, but what happens if someone send thousand of invalid headers in short time as an attack ? we would be trying to process them without success, wouldn't that be the kind of thing we try to avoid by adding the limits?